source/Firmata/UwpFirmata.cpp

```diff
@@ -333,6 +348,35 @@ UwpFirmata::inputThread(
     }
 }
 
+void
+UwpFirmata::onConnectionEstablished(
+    void
+    )
+{
+    {   //critical section guarantees state is only changed when it is not being modified elsewhere
```

Comment:
This protection might not do what you think it does - You are going to trounce _connection_ready no matter what its current state is. Perhaps you meant InterlockedCompareExchange?